test/fw/ptl/utils/pbs_anonutils.py

```diff
@@ -98,6 +100,96 @@ def __init__(self, attr_delete=None, resc_delete=None,
         self.gmap_attr_key = {}
         self.gmap_resc_key = {}
 
+    def __get_kvl(self, text):
+        """
+        Extract key-value pairs from the text specified
+
+        :param text: the line of text to extract key-value pairs from
+        :type text: String
+
+        :returns: a dict of key value pairs
+        """
+        if text is None:
+            return None
+
+        if "=" not in text:
+            # No key-value pairs in the text
+            return None
+
+        # Split the text by spaces
+        list_items = shlex.split(text)
+
+        kvl = {}
+
+        # Go through the list and handle the following two cases:
+        #     "key = value"
+        #     "key=value"
+        index = 0
+        for item in list_items:
+            if item == "=":
+                # The item before and after will be key-value pair
+                key = list_items[index - 1]
+                value = list_items[index + 1]
+                kvl[key] = value
+            elif "=" in item:
+                delim = "="
+                # qmgr print outputs can have key += value or key -= value
+                if "+=" in item:
+                    delim = "+="
+                elif "-=" in item:
+                    delim = "-="
+
+                key, value = item.split(delim)
+                kvl[key] = value
+            index += 1
+        return kvl
+
+    def __get_anon_key(self, key, map):
+        """
+        Get an anonymized string for the 'key' belonging to the map provided
+
+        :param key: the key to anonymize
+        :type key: String
+        :param map: the map to which the key belongs
+        :type map: dict
+
+        :returns: an anonymized string for the key
+        """
+        if key in map.keys():
+            anon_key = map[key]
+        else:
+            anon_key = self.utils.random_str(len(key))
+            map[key] = anon_key
+
+        return anon_key
+
+    def __get_anon_value(self, key, value, map):
+        """
+        Get an anonymied string for the 'value' belonging to the map provided
+        The map will be in the following format:
+            key:{val1:anon_val1, val2:anon_val2, ...}
+
+        :param key: the key for this value
+        :type key: String
+        :param value: the value to anonymize
+        :type value: String
+        :param map: the map to which the key belongs
+        :type map: dict
+
+        :returns: an anonymized string for the value
+        """
+
+        if key == "project" and value == "_pbs_project_default":
+            return
+        if key in map:
+            value_map = map[key]
+            anon_val = self.__get_anon_key(value, value_map)
```

Comment:
You don't really need value_map.  You can just call __get_anon_key() with map[key]